### PR TITLE
Update Add to cart function

### PR DESCRIPTION
**Pull request by TuyenNguyen552**

### CODE REVIEW

File: src/test/java/com/qmetry/qaf/example/steps/TC03.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import com.qmetry.qaf.automation.ui.webdriver.QAFExtendedWebElement;
+import com.qmetry.qaf.example.pages.CategoryPage;
+import com.qmetry.qaf.example.pages.HomePage;
+import com.qmetry.qaf.example.pages.Login;
+
+public class TC03 extends StepsLibrary {
+
+	HomePage homePage = new HomePage();
+	Login loginPage = new Login();
+	Properties prop = new Properties();
```
Comment (truongtuanphat) [inline src/test/java/com/qmetry/qaf/example/steps/TC03.java:21]: We don't use this one, please kindly remove it

File: src/test/java/com/qmetry/qaf/example/steps/TC03.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+	CategoryPage categoryPage = new CategoryPage();
+
+    @QAFTestStep(description = "I will be show the {0} page")
+    public void iWillBeShowThePage(String page) throws Exception {
+    	 String xpath = String.format("//*[@class='navigation_page' and contains(text(), '%s')]", page);
```
Comment (truongtuanphat) [inline src/test/java/com/qmetry/qaf/example/steps/TC03.java:26]: If we navigate to Women -> Dresses, the structure of tags will be changed and this one won't work anymore.
I suggest using title of website to verify that, example "Women - My Store", "Dresses - My Store"

File: src/test/java/com/qmetry/qaf/example/pages/CategoryPage.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+import com.qmetry.qaf.automation.ui.webdriver.QAFExtendedWebElement;
+import com.qmetry.qaf.example.steps.StepsLibrary;
+
+public class CategoryPage extends StepsLibrary {
+	private String xpath = "";
+	
+	 
+	public String getDialogContent() {
+		xpath = "//*[@id='layer_cart' and not(contains(@style,'display: none'))]";
```
Comment (truongtuanphat) [inline src/test/java/com/qmetry/qaf/example/pages/CategoryPage.java:11]: Please kindly move this xpath to locator file and also remove which ones are not used

File: src/test/java/com/qmetry/qaf/example/steps/TC03.java [hunk 4]
```diff
@@ -2,32 +2,32 @@
...
 
 	HomePage homePage = new HomePage();
-	Login loginPage = new Login();
-	Properties prop = new Properties();
 	CategoryPage categoryPage = new CategoryPage();
-
+		
+	//* Button: Add to card, More, Quick view
+    @QAFTestStep(description = "I click on {button} button of the product as {productName}")
+    public void clickOnButton(String button, String productName ) throws Exception {
```
Comment (truongtuanphat) [inline src/test/java/com/qmetry/qaf/example/steps/TC03.java:18]: Change method name to iClickOnButtonOfTheProductAsProductName

File: src/test/java/com/qmetry/qaf/example/steps/TC03.java [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+	HomePage homePage = new HomePage();
+	CategoryPage categoryPage = new CategoryPage();
+		
+	//* Button: Add to card, More, Quick view
+    @QAFTestStep(description = "I click on {button} button of the product as {productName}")
+    public void clickOnButton(String button, String productName ) throws Exception {
+    	categoryPage.moveToElement(productName);
+        String xpath = String.format("//ul[contains(@class,'product_list')]/li[contains(@class,'hovered')]//span[text()='%s']", button);
+        QAFExtendedWebElement ele = new QAFExtendedWebElement(xpath);
+        waitUntilElementExisted(ele);
```
Comment (truongtuanphat) [inline src/test/java/com/qmetry/qaf/example/steps/TC03.java:22]: clickOnElement already included waiting process. No need to use waitUntilElementExisted, you should try.

File: src/test/java/com/qmetry/qaf/example/steps/TC03.java (hunk 4 above)
Comment (TuyenNguyen552) [reply src/test/java/com/qmetry/qaf/example/steps/TC03.java]: Fixed

File: src/test/java/com/qmetry/qaf/example/steps/TC03.java (hunk 5 above)
Comment (TuyenNguyen552) [reply src/test/java/com/qmetry/qaf/example/steps/TC03.java]: Fixed

Comment (truongtuanphat) [reply src/test/java/com/qmetry/qaf/example/steps/TC03.java]: Approved

File: src/test/java/com/qmetry/qaf/example/steps/TC03.java (hunk 4 above)
Comment (truongtuanphat) [reply src/test/java/com/qmetry/qaf/example/steps/TC03.java]: Approved

File: src/test/java/com/qmetry/qaf/example/steps/TC03.java (hunk 1 above)
Comment (TuyenNguyen552) [reply src/test/java/com/qmetry/qaf/example/steps/TC03.java]: Fixed

File: src/test/java/com/qmetry/qaf/example/steps/TC03.java (hunk 2 above)
Comment (TuyenNguyen552) [reply src/test/java/com/qmetry/qaf/example/steps/TC03.java]: Fiexd

File: src/test/java/com/qmetry/qaf/example/pages/CategoryPage.java (hunk 3 above)
Comment (TuyenNguyen552) [reply src/test/java/com/qmetry/qaf/example/pages/CategoryPage.java]: Fixed

File: src/test/java/com/qmetry/qaf/example/steps/TC03.java (hunk 1 above)
Comment (truongtuanphat) [reply src/test/java/com/qmetry/qaf/example/steps/TC03.java]: Approved

File: src/test/java/com/qmetry/qaf/example/steps/TC03.java (hunk 2 above)
Comment (truongtuanphat) [reply src/test/java/com/qmetry/qaf/example/steps/TC03.java]: Approved

File: src/test/java/com/qmetry/qaf/example/pages/CategoryPage.java (hunk 3 above)
Comment (truongtuanphat) [reply src/test/java/com/qmetry/qaf/example/pages/CategoryPage.java]: Approved

**Review by truongtuanphat — APPROVED** [review]
Approved

**Pull request merged**